packages/snap/src/index.ts

```diff
@@ -143,29 +141,62 @@ async function sendTransaction(tx: CeloTransactionRequest, wallet: CeloWallet) {
   return txResponse.wait()
 }
 
-async function getBIP44Node(index: number = 0): Promise<BIP44Node> {
-  // Get the bip44 node corresponding to the path m/44'/52752'/0'/0
-  const bip44Node = await snap.request({
+async function getKeyPair(snap: SnapsGlobalObject, address?: string, addressIndex: number = 0): Promise<KeyPair> {
+  const derivationPath = "m/44'/60'/0'/0"; //Todo - read from state config
+  let extendedPrivateKey;
+  let _addressIndex = 0;
+  const MAX_SEARCH_DEPTH = 50;
+
+  const [, , coinType, account, change] = derivationPath.split('/');
+  const bip44Code = coinType.replace("'", '');
+
+  const bip44Node = (await snap.request({
     method: 'snap_getBip44Entropy',
     params: {
-      coinType: 52752, // https://github.com/satoshilabs/slips/blob/master/slip-0044.md
+      coinType: Number(bip44Code),
     },
-  })
-  const deriveAddress = await getBIP44AddressKeyDeriver(bip44Node)
-  // Derive account w index 
-  return deriveAddress(index)
-}
+  })) as JsonBIP44CoinTypeNode;
 
-async function getPrivateKey(bip44Node?: BIP44Node, index: number = 0): Promise<string> {
-  if (!bip44Node) {
-    bip44Node = await getBIP44Node(index)
-  }
-  if (!bip44Node.privateKey) {
-    throw new Error('Private key is undefined. BIP-44 node is public.')
+  const addressKeyDeriver = await getBIP44AddressKeyDeriver(bip44Node, {
+    account: parseInt(account),
+    change: parseInt(change),
+  }); 
+
+  if (address) {
+    let search;
+    do {
+      search = await addressKeyDeriver(Number(_addressIndex));
+      if (search.address.toLowerCase() == address.toLowerCase()) {
+        extendedPrivateKey = search
+      }
+      _addressIndex = _addressIndex + 1;
+    } while (
+      address.toLowerCase() != search.address.toLowerCase() &&
+      _addressIndex <= MAX_SEARCH_DEPTH
+    );
+  } else {
+    extendedPrivateKey = await addressKeyDeriver(Number(addressIndex));
+  } 
+
+  if (!extendedPrivateKey) {
+    await snap.request({
+      method: 'snap_dialog',
+      params: {
+        type: 'alert',
+        content: panel([
+          text(`Oops we could not locate the private key to complete the transaction!`),
```

Comment:
suggestion

`The Transaction specifies from {address} however that address could not be signed by the private key derived from MetaMask `